README.org

```diff
@@ -371,19 +373,163 @@ We can also change the thickness of the lines and shapes that we draw by changin
 #+END_SRC
 
 *** Transforms
-TODO
+The transforms =(translate dx dy)=, =(rotate angle &optional (cx 0) (cy 0))= and =(scale sx &optional sy (cx 0) (cy 0))= are available to change the view matrix that is applied to coordinates.
+
+The current view can be saved on a stack and restored with =(push-matrix)= and =(pop-matrix)=, which are analogous to =push()= and =pop()= in p5.js. The macro =(with-identity-matrix &body body)= pushes the current view matrix onto the stack, sets the view matrix to the identity matrix, executes =body=, and then pops the view matrix. =(with-current-matrix &body body)= is the same, except it doesn't change the view matrix after pushing it.
+
+In this example, translation and rotation are used to draw a triangle in the centre of the screen, without explicitly defining the coordinates of the vertices.
+
+#+BEGIN_SRC lisp
+  (defsketch transform-test
+    ((title "Transform test")
+     (width 500)
+     (height 500)
+     (side 100)
+     (y-offset (/ side (* 2 (tan (radians 60))))))
+   (with-identity-matrix
+     (translate 250 250)
+     (loop repeat 3
+           do (line (- (* 1/2 side)) y-offset  (* 1/2 side) y-offset)
+           do (rotate 120))))
+#+END_SRC
+
+This example draws a sequence of increasingly shrinking squares using scaling.
+
+#+BEGIN_SRC lisp
+  (defsketch transform-test
+     ((width 400)
+	  (height 400)
+	  (title "Scale test"))
+   (translate 100 100)
+   (dotimes (x 5)
+     (rect 0 0 100 100)
+     (translate 150 0)
+	 (scale 1/2)))
+#+END_SRC
 
 *** Text
-TODO
+Use =(text text-string x y &optional width height)= to draw text, where =x= and =y= specify the top-left corner of the rectangle containing the text. =width= and =height= control the shape of the text box. There is support for changing the [[https://github.com/vydd/sketch/blob/master/src/font.lisp#L29][font]].
+
+#+BEGIN_SRC lisp
+  (defsketch text-test
+     ((title "Hello, world!"))
+   (text title 0 0 100))
+#+END_SRC
 
 *** Images
-TODO
+First =(load-resource filename ...)= to load the image from a given file, then =(image image-resource x y &optional width height)= to draw the image with its top-left corner at =(x, y)= and with the given =width= and =height=.
+
+#+BEGIN_SRC lisp
+  (defsketch image-test
+     ((title "Hello, image!")
+      (rsc (load-resource "/path/to/img.png")))
+    (image rsc 10 10 200 200))
+#+END_SRC
 
-*** Bindings
-TODO
```

Comment:
Hm, seems to be deleted by accident?

Reply:
I deleted this because I wasn't sure what it was referring to. I also think it makes more sense to create an issue to track missing docs, rather than leaving TODOs that people will forget about.